src/WorkBC.Data/Migrations/20240424212524_UpdateNocCodes2021Data.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+using System.Text.Json;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System;
+using System.IO;
+using System.Text;
+
+#nullable disable
+
+namespace WorkBC.Data.Migrations
+{
+    public partial class UpdateNocCodes2021Data : Migration
+    {
+        public UpdateNocCodes2021Data() : base()
+        {
+
+        }
+        protected async override void Up(MigrationBuilder migrationBuilder)
+        {
+
+            var allNocs = GetNocCodes2021();
+
+            foreach (var noc in allNocs)
+            {
+                if (!String.IsNullOrEmpty(noc.noc_2021))
+                {
+                    //The field 'Id' is derived from the field 'Code' and is not an auto-generated Identity field.
+                    int.TryParse(noc.noc_2021.ToString(), out int id);
+
+                    string code = noc.noc_2021.ToString();
+                    string title = noc.label.ToString();
+                    string frenchTitle = noc.label_fr.ToString();
+
+                    //Writing the data to the table dbo.NocCodes2021
+                    migrationBuilder.InsertData(
+                        table: "NocCodes2021",
+                        columns: new[] { "Id", "Code", "Title", "FrenchTitle" },
+                        values: new object[] { id, code, title, frenchTitle });
+
+                }
+
+            }
+
+        }
+
+        protected async override void Down(MigrationBuilder migrationBuilder)
+        {
+
```

Comment:
Move the truncate instruction here.

Reply:
Cannot move the truncate here since FK association will fail. Added delete command instead.